services-bytebuf-codec/src/main/java/io/scalecube/services/transport/api/BinaryHeadersCodec.java

```diff
@@ -0,0 +1,90 @@
+package io.scalecube.services.transport.api;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.ByteBufOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Simple binary codec for headers service message.
+ */
+public class BinaryHeadersCodec implements HeadersCodec {
```

Comment:
Name of this class must somehow reflect that this's pure java implementation. Like `HeadersCodecImpl`, `DefaultHeadersCodec` .. suggest yours. `Binary..Codec`  doesn't reflect a fact that this is default out-of-the-box headers codec which doesn't require any additional dependencies.

Reply:
the goal was not to write the codec on pure java, but to replace json with a binary format with a minimum overhead in size.